components/style/values/specified/mod.rs

```diff
@@ -728,6 +728,51 @@ impl LengthOrPercentageOrNumber {
     }
 }
 
+/// <length> | <number> | <percentage> for SVG stroke-*
+#[allow(missing_docs)]
+#[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+#[derive(Clone, Debug, PartialEq, ToCss)]
+pub enum SVGStrokeLength {
+    LengthOrPercentage(LengthOrPercentage),
+    Number(Number),
+}
+
+no_viewport_percentage!(SVGStrokeLength);
```

Comment:
Well, it can have viewport units, right? If so, this shouldn't really be `no_viewport_percentage!`. You can `derive(HasViewportPercentage)`, I think.